### PR TITLE
refactor: ♻️ pipeline conversion function

### PR DESCRIPTION
**Pull request by slugb0t**

### CODE REVIEW

File: .github/workflows/publish-to-pypi-beta.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
   push:
     branches:
       - staging
+      - conversion-function
```
Comment (megasanjay) [inline .github/workflows/publish-to-pypi-beta.yml:9]: What is this for?

File: pyproject.toml [hunk 2]
```diff
@@ -39,7 +39,7 @@ dicttoxml = "^1.7.16"
 types-requests = "^2.30.0.0"
 pymdown-extensions = "^10.0.1"
 validators = "^0.20.0"
-art = "^6.0"
+art = "^6.0"_
```
Comment (megasanjay) [inline pyproject.toml:42]: stray `_`?

File: pyfairdatatools/utils.py [hunk 3]
```diff
@@ -47,6 +51,309 @@ def validate_file_path(file_path, preexisting_file=False, writable=False):
...
 
     return False if isinstance(result, ValidationFailure) else result
+
+
+def generate_random_identifier(k):
+    """Generate a random identifier"""
+    return "".join(random.choices(string.ascii_lowercase + string.digits, k=k))
+
+
+def convert_for_datasite(data):
```
Comment (megasanjay) [inline pyfairdatatools/utils.py:65]: datasite => datacite

File: .github/workflows/publish-to-pypi-beta.yml (hunk 1 above)
Comment (megasanjay) [reply .github/workflows/publish-to-pypi-beta.yml:9]: Should be removed before PR merges

File: tests/test_validate.py [hunk 4]
```diff
@@ -609,6 +601,35 @@ def test_language(self):
...
+        """Test valid dataset description."""
+        data = deepcopy(self.valid_data)
+
+        try:
+            output = validate_dataset_description(data)
+        except Exception as e:  # pylint: disable=broad-except
+            print(e)
+            output = False
+
+        assert output
```
Comment (megasanjay) [inline tests/test_validate.py:631]: I think this should be tested first. Ensures that the correct one passes

**Review by megasanjay — CHANGES_REQUESTED** [review]
Minor fixes

File: .github/workflows/publish-to-pypi-beta.yml (hunk 1 above)
Comment (slugb0t) [reply .github/workflows/publish-to-pypi-beta.yml:9]: I was trying to publish a beta build to test but it wasn't working due to the yml settings

**Review by megasanjay — APPROVED** [review]

**Pull request merged**